.github/workflows/ci.yml

```diff
@@ -1,11 +1,6 @@
 name: TileDB Python CI
 
-on: 
-  push:
-    branches: [dev]
-  pull_request:
-    branches: [dev]
```

Comment:
I think that we should just remove this condition for `pull_request` but leave other untoched

Reply:
After some consideration and a small chat with Agis I have decided that sometimes it is also necessary to run CI on every branch, even without PR.